pkg/server/server.go

```diff
@@ -391,156 +550,14 @@ func (s *Server) Run(ctx context.Context) error {
 	// TODO: split apart everything after this line, into their own commands, optional launched in this process
 
 	controllerConfig := rest.CopyConfig(s.IdentityConfig)
+	controllerConfig.Timeout = time.Second * 30
 
-	if err := s.installOpenAPIv3Controller(ctx, controllerConfig); err != nil {
-		return err
-	}
-
-	if err := s.installKubeNamespaceController(ctx, controllerConfig); err != nil {
-		return err
-	}
-
-	if err := s.installClusterRoleAggregationController(ctx, controllerConfig); err != nil {
-		return err
-	}
-
-	if err := s.installKubeServiceAccountController(ctx, controllerConfig); err != nil {
-		return err
-	}
-
-	if err := s.installKubeServiceAccountTokenController(ctx, controllerConfig); err != nil {
-		return err
-	}
-
-	if err := s.installRootCAConfigMapController(ctx, s.Apis.GenericAPIServer.LoopbackClientConfig); err != nil {
-		return err
-	}
-
-	if err := s.installApiExportIdentityController(ctx, controllerConfig); err != nil {
+	// Adding the bootup time controllers and informers initialization
+	s.controllerLock.Lock()
```

Comment:
any reason why we should move the locking into install/uninstallControllers? Looks like the lock is only needed there.

Reply:
removed mutex locks